tests/0004-conf.c

```diff
@@ -498,6 +498,37 @@ static void do_message_timeout_linger_checks(void) {
         SUB_TEST_PASS();
 }
 
+/**
+ * @brief Verify behaviour of checking that message.timeout.ms fits within
+ *        configured transaction.timeout.ms. By jainruchir.
+ */
+static void do_message_timeout_transaction_timeout_checks(void) {
+        rd_kafka_conf_t *conf;
+        rd_kafka_t *rk;
+        char numstr[8];
+        char errstr[512];
+        const char *transactional_id = "txnid";
+
+        SUB_TEST_QUICK();
+        test_conf_init(&conf, NULL, 60);
+        test_conf_set(conf, "transactional.id", transactional_id);
+        test_conf_set(conf, "socket.connection.setup.timeout.ms", "5000");
```

Comment:
socket.connection.setup.timeout.ms and reconnect.backoff.max.ms are not needed by this test